ath9k.md

```diff
@@ -0,0 +1,24 @@
+---
+title: ath9k
+nav_order: 2
+---
+
+# `ath9k`
+
+## Overview
+
+The kernel network driver used throughout the project is `ath9k`. This page intends to provide an overview of the underlying logic of this driver. 
+
+The source code of the `ath9k` driver can be found here at [Elixir Bootlin](https://elixir.bootlin.com/linux/latest/source/drivers/net/wireless/ath/ath9k).
+
+## Transmission
+
+In this driver, there exists a linked list of empty packet nodes (`tx_buf`), where each node is a C `struct` containing space to record the information of one to-be-transmitted packet. 
+
+When `ath9k` receives a request to send a packet, it tries to obtain (pop) an empty node from the linked list buffer (`tx_buf`), fill in the packet information into the node, and then pass this node to a transmission routine by appending it to a transmission queue. After the transmission is finished, the node is emptied and gets added back to the buffer. Since the linked list buffer has a limited size (which can be obtained using the `ATH_TXBUF` macro) and each transmission temporarily pops one of its node, the linked list can be empty under high volumn of transmission demand. In such case, a packet is dropped thus cannot be transmitted.
```

Comment:
"a packet is dropped thus cannot be transmitted."
"a packet is dropped and thus cannot be transmitted."